server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -621,13 +628,20 @@ private void stopBeingLeader()
     return ImmutableList.copyOf(duties);
   }
 
-  public class DutiesRunnable implements Runnable
+  private List<CoordinatorDuty> makeCompactSegmentsDuty()
+  {
+    List<CoordinatorDuty> duties = new ArrayList<>();
+    duties.add(compactSegments);
+    return ImmutableList.copyOf(duties);
```

Comment:
Why not `return ImmutableList.of(compactSegments);` instead?

Reply:
Done